src/theory/arrays/theory_arrays_rewriter.h

```diff
@@ -371,7 +371,7 @@ class TheoryArraysRewriter {
               elements.pop_back();
             }
             Trace("arrays-postrewrite") << "Arrays::postRewrite returning " << n << std::endl;
-            return RewriteResponse(REWRITE_DONE, n);
+            return RewriteResponse((n == node) ? REWRITE_DONE : REWRITE_AGAIN, n);
```

Comment:
I suppose this is fine assuming the rest of the rules are terminating. Is there a tighter condition we can test here or in an assert if it is too inefficient?